modules/101-cert-manager/templates/cert-manager/clusterissuer-cloudflare.yaml

```diff
@@ -20,7 +20,14 @@ spec:
     - dns01:
         cloudflare:
           email: "{{ .Values.certManager.cloudflareEmail }}"
+          {{- if and (hasKey .Values.certManager "cloudflareGlobalAPIKey") (.Values.certManager.cloudflareGlobalAPIKey) }}
           apiKeySecretRef:
             name: cloudflare
             key: global-api-key
+          {{- end }}
+          {{- if and (hasKey .Values.certManager "cloudflareAPIToken") (.Values.certManager.cloudflareAPIToken) }}
+          apiTokenSecretRef:
```

Comment:
How does it work if a user provides both keys? I assume cert-manager uses api-token first and then falls back to the global API key, but it is just an assumption.

Reply:
Is it possible to validate it using openapi module spec? (in case someone wants to specify both keys in config)

Reply:
I don't see the way. Because openapi spec for cert-manager contains a lot of parameters for other providers. We can't use anyOf/oneOf here. Maybe I'm missing something